File: MonteCarlo/mc.scala
import org.apache.spark.SparkContext
import math
import java.io._

object MonteCarloPI {
  def main(args: Array[String]): Unit = {
    val NUM_SAMPLES: Int = 100000000
    val sc = new SparkContext()


    val log = new PrintWriter(new File("montecarlo-scala.log" ))

    for(_ <- 1 to 100) {
      val t0: Long = System.currentTimeMillis / 1000

      val count = sc.parallelize(1 to NUM_SAMPLES).filter { _ =>
        val x = math.random
        val y = math.random
        x*x + y*y < 1
      }.count()

      log.write(
        s"Elapsed time: ${System.currentTimeMillis / 1000 - t0}\n")
      log.write(s"Pi is roughly ${4.0 * count / NUM_SAMPLES}\n")
    }
    log.close()
  }

File: MonteCarlo/python/mc.py
from __future__ import print_function

import random
import time

from pyspark import SparkContext


if __name__ == "__main__":
    NUM_SAMPLES = 100000000
    sc = SparkContext()
    with open("montecarlo-py.log","w") as log:
        for _ in range(10):

            def inside(p):
                x, y = random.random(), random.random()
                return x*x + y*y < 1

            t0 = time.time()

            count = sc.parallelize(range(0, NUM_SAMPLES)) \
                .filter(inside).count()
            print("Elapsed time: %f" % (time.time()-t0), file=log)
            print("Pi is roughly: %f" % (4.0*count / NUM_SAMPLES), \
                    file=log)
    sc.stop()
